galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java

```diff
@@ -145,21 +146,46 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
 
       Map<String, String[]> query = req.getParameterMap();
 
+      //Checking ascending or descending for sorting
+      
       if(!query.isEmpty()){
+         
          if(!ExtractQuerySort.isAscending(query, "to")) {
             Collections.reverse(runs);
          }
+         
+         boolean testClassSort = ExtractQuerySort.isAscending(query,"testclass");
+         
+         if(testClassSort) {
```

Comment:
The logic of this needs to be better.   you need if, else if, else   to look at all the sort options in one go.  not as separate entities